app/flatpak-builtins-alias.c

```diff
@@ -0,0 +1,233 @@
+          if (flatpak_dir_is_user (dir))
+            {
+              g_ptr_array_insert (dirs, 0, g_object_ref (dir));
+              g_ptr_array_remove_index (dirs, i);
+              break;
+            }
+        }
+    }
+
+  if (argc == 1)
+    {
+      g_autoptr(FlatpakTablePrinter) printer = NULL;
+
+      printer = flatpak_table_printer_new ();
+      flatpak_table_printer_set_column_title (printer, 0, _("Alias"));
+      flatpak_table_printer_set_column_title (printer, 1, _("App"));
+      flatpak_table_printer_set_column_title (printer, 2, _("Installation"));
+
+      for (i = 0; i < dirs->len; i++)
+        {
+          FlatpakDir *dir = g_ptr_array_index (dirs, i);
+          g_autoptr(GHashTable) aliases = NULL; /* alias → app-id */
+
+          aliases = flatpak_dir_get_aliases (dir);
+
+          GLNX_HASH_TABLE_FOREACH_KV (aliases, const char *, alias, const char *, app_id)
+            {
+              flatpak_table_printer_add_column (printer, alias);
+              flatpak_table_printer_add_column (printer, app_id);
+              flatpak_table_printer_add_column (printer, flatpak_dir_get_name_cached (dir));
+              flatpak_table_printer_finish_row (printer);
+            }
+        }
+
+      if (flatpak_table_printer_get_current_row (printer) > 0)
+        flatpak_table_printer_print (printer);
+      else if (flatpak_fancy_output ())
+        g_print (_("No aliases\n"));
+    }
+  else if (opt_remove && argc == 2)
+    {
+      const char *alias = argv[1];
+      g_autoptr(GError) saved_error = NULL;
+
+      for (i = 0; i < dirs->len; i++)
+        {
+          g_autoptr(GError) local_error = NULL;
+          FlatpakDir *dir = g_ptr_array_index (dirs, i);
+
+          if (flatpak_dir_remove_alias (dir, alias, &local_error))
+            return TRUE;
+          else
+            {
+              if (g_error_matches (local_error, FLATPAK_ERROR, FLATPAK_ERROR_ALIAS_NOT_FOUND))
+                {
+                  if (saved_error == NULL)
+                    g_propagate_error (&saved_error, g_steal_pointer (&local_error));
+                  continue;
+                }
+              else
+                {
+                  g_propagate_error (error, g_steal_pointer (&local_error));
+                  return FALSE;
+                }
+            }
+        }
+
+      g_propagate_error (error, g_steal_pointer (&saved_error));
+      return FALSE;
+    }
+  else if (!opt_remove && argc == 3)
+    {
+      const char *alias = argv[1];
+      const char *app = argv[2];
+      FlatpakDir *dir_to_use = NULL;
+      g_autoptr(FlatpakDecomposed) current = NULL;
+
+      /* Check if the named app is deployed */
+      for (i = 0; i < dirs->len; i++)
+        {
+          FlatpakDir *dir = g_ptr_array_index (dirs, i);
+          g_autoptr(GFile) deploy = NULL;
+
+          g_clear_pointer (&current, flatpak_decomposed_unref);
+          current = flatpak_dir_current_ref (dir, app, cancellable);
```

Comment:
I'm not sure this is always right. Even if the app is from system, that doesn't mean you necessarily want to make a system-global alias. It seems to me like aliases are primarily a per-user thing, and only for special cases they are global.